orchestration/dagster/hca_orchestration/resources/base.py

```diff
@@ -0,0 +1,165 @@
+class DataflowBeamRunner:
+    def __init__(self, project, temp_location, subnet_name, service_account, image_name, image_version, namespace):
+        self.project = project
+        self.temp_location = temp_location
+        self.subnet_name = subnet_name
+        self.service_account = service_account
+        self.image_name = image_name
+        self.image_version = image_version
+        self.namespace = namespace
+
+    def run(self, job_name, input_prefix, output_prefix, context):
+        args = [
+            '--runner=dataflow',
+            f"--inputPrefix={input_prefix}",
+            f"--outputPrefix={output_prefix}",
+            f"--project={self.project}",
+            f"--region=us-central1",
+            f"--tempLocation={self.temp_location}",
+            f"--subnetwork=regions/us-central1/subnetworks/{self.subnet_name}",
+            f"--serviceAccount={self.service_account}",
+            f"--workerMachineType=n1-standard-4",
+            f"--autoscalingAlgorithm=THROUGHPUT_BASED",
+            f"--numWorkers=4",
+            f"--maxNumWorkers=16",
+            f"--experiments=shuffle_mode=service"
+        ]
+
+        image_name = f"{self.image_name}:{self.image_version}"  # {context.solid_config['version']}"
+        job = self.dispatch_k8s_job(self.namespace, image_name, job_name, args, context)
+        context.log.info(f"Dataflow job started")
+
+        self.get_job_status(job.metadata.name, self.namespace)
+        client = DagsterKubernetesClient.production_client()
+        client.wait_for_job_success(job.metadata.name, self.namespace)
+
+    @staticmethod
+    def get_job_status(name, namespace):
+        client = kubernetes.client.BatchV1Api()
+        return client.read_namespaced_job_status(name, namespace)
+
+    @staticmethod
+    def dispatch_k8s_job(namespace, image_name, job_name_prefix, args, context):
+        # we will need to poll the pod/job status on creation
+        kubernetes.config.load_kube_config()
+
+        job_name = f"{job_name_prefix}-{uuid4()}"
+        pod_name = f"{job_name}-pod"
+        job_container = kubernetes.client.V1Container(
+            name=job_name,
+            image=image_name,
+            args=args,
+        )
+
+        # TODO unhardcode the SA name below
+        template = kubernetes.client.V1PodTemplateSpec(
+            metadata=kubernetes.client.V1ObjectMeta(name=pod_name),
+            spec=kubernetes.client.V1PodSpec(
+                restart_policy="Never",
+                containers=[job_container],
+                service_account_name="argo-runner"
+            ),
+        )
+
+        job = kubernetes.client.V1Job(
+            api_version="batch/v1",
+            kind="Job",
+            metadata=kubernetes.client.V1ObjectMeta(
+                name=job_name
+            ),
+            spec=kubernetes.client.V1JobSpec(
+                template=template,
+                backoff_limit=2,
+                ttl_seconds_after_finished=86400
+            ),
```

Comment:
this looks like a lot of boilerplate, good candidate to pull out into a helper method later, i think.